src/lib/containers/SqlEditor.tsx

```diff
@@ -0,0 +1,84 @@
+import { library } from '@fortawesome/fontawesome-svg-core'
+import { faSearch } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { Collapse } from '@material-ui/core'
+import React, { useEffect, useRef, useState } from 'react'
+import {
+  QueryWrapperChildProps,
+  QUERY_FILTERS_COLLAPSED_CSS,
+  QUERY_FILTERS_EXPANDED_CSS,
+  TopLevelControlsState,
+} from './QueryWrapper'
+
+library.add(faSearch)
+
+export type SqlEditorProps = Pick<
+  QueryWrapperChildProps,
+  'executeQueryRequest' | 'getLastQueryRequest'
+> & {
+  topLevelControlsState?: Pick<
+    TopLevelControlsState,
+    'showSqlEditor' | 'showFacetFilter'
+  >
+}
+
+export function SqlEditor(props: SqlEditorProps) {
+  const [sql, setSql] = useState('')
+  const inputRef = useRef<HTMLInputElement>(null)
+  const showSqlEditor = props.topLevelControlsState?.showSqlEditor
+  const showFacetFilter = props.topLevelControlsState?.showFacetFilter
+  const {getLastQueryRequest} = props
+  useEffect(() => {
+    if (showSqlEditor) {
+      const defaultSql = getLastQueryRequest ? getLastQueryRequest().query.sql : ''
+      setSql(defaultSql)
+      inputRef.current?.focus()
+    }
+  }, [showSqlEditor, getLastQueryRequest])
+
+  const search = (event: React.SyntheticEvent<HTMLFormElement>) => {
+    event.preventDefault()
+    const { executeQueryRequest, getLastQueryRequest } = props
+    const lastQueryRequestDeepClone = getLastQueryRequest!()
+    lastQueryRequestDeepClone.query.sql = sql
+
+    executeQueryRequest!(lastQueryRequestDeepClone)
+  }
+
+  const handleChange = (event: React.FormEvent<HTMLInputElement>) => {
+    inputRef.current?.setCustomValidity('')
+    setSql(event.currentTarget.value)
+  }
+
+  return (
+    <div
+      className={`SearchV2 ${
```

Comment:
SQLEditor adopts same styling as the search boxes.

Reply:
Suggest making the CSS class name more generic, like `QueryWrapperTextInput`, so it's clear that the class isn't tightly coupled with the SearchV2 component

Reply:
Great idea, will do